android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/notifications/NotificationsModule.java

```diff
@@ -156,7 +156,7 @@ public void getExponentPushTokenAsync(final Promise promise) {
     }
 
     try {
-      String experienceId = mManifest.getString(ExponentManifest.MANIFEST_ID_KEY);
+      String experienceId = ExponentManifest.getExperienceId(mManifest);
```

Comment:
is the push notification server going to be set up to handle possibly taking the `scopeKey` as the experienceId?

Reply:
@tcdavis / @ide - this one is for you

Reply:
We haven't planned any backend changes other than adding the two fields to the manifest.
Is that required to avoid invalidating push tokens when a project is transferred?

Reply:
No, the notification server expects the legacy experience name. Access tokens depend on this for now until we can fully migrate to project IDs.

Reply:
That means we also need to change iOS to use the `manifest.id` field for getting push tokens, correct? Since `experienceId` will now map to the scope key

Reply:
We need to send down a new field with manifests that specifies the project's ID (a UUID) and change the clients to send that to our servers when registering a device for push notifications.

We will have four fields in the manifest:
* `id`: the legacy experience ID. This will be `@originaluser/originalslug` for old projects and `@currentuser/currentslug` for new projects. Don't use this except for legacy purposes.
* `projectName`: the developer-facing string that will be `@currentuser/currentslug`.
* `scopeKey`: an immutable value for a project that is intended for scoping. This will be a string like `sk-${nanoid()}` for production projects hosted by Expo's application services, but it can be any unique string. This will likely be required whether or not the developer is using Expo's application services. Also, this value will differ between development and production (XDL will use `@currentuser/currentslug` for the scope key.).
* `expoProjectId`: an immutable value that is used to identify a project hosted by Expo's application services. It is the UUID PK of the project. This is NOT used for scoping. It will be specified only for projects hosted by Expo's application services. XDL and other servers will not specify it (after all, they are not using Expo's services so there is no Expo project ID nor do they need to identify an Expo project).